futures-util/src/compat/compat01as03.rs

```diff
@@ -52,15 +61,49 @@ pub trait Stream01CompatExt: Stream01 {
     /// [`Stream<Item = T, Error = E>`](futures::stream::Stream)
     /// into a futures 0.3
     /// [`Stream<Item = Result<T, E>>`](futures_core::stream::Stream).
-    fn compat(self) -> Compat01As03<Self> where Self: Sized {
+    fn compat(self) -> Compat01As03<Self>
+    where
+        Self: Sized,
+    {
         Compat01As03::new(self)
     }
 }
 impl<St: Stream01> Stream01CompatExt for St {}
 
-fn poll_01_to_03<T, E>(x: Result<Async01<T>, E>)
-    -> task03::Poll<Result<T, E>>
-{
+/// Extension trait for futures 0.1 [`Sink`](futures::sink::Sink)
+pub trait Sink01CompatExt: Sink01 {
+    /// Converts a futures 0.1
+    /// [`Sink<SinkItem = T, SinkError = E>`](futures::sink::Sink)
+    /// into a futures 0.3
+    /// [`Sink<SinkItem = T, SinkError = E>`](futures_sink::sink::Sink).
+    fn compat(self) -> Compat01As03Sink<Self, Self::SinkItem>
+    where
+        Self: Sized,
+    {
+        Compat01As03Sink::new(self)
+    }
+}
+impl<Si: Sink01> Sink01CompatExt for Si {}
+
+/// Extension trait for futures 0.1 [`Stream`](futures::stream::Stream) and
+///  [`Sink`](futures::sink::Sink) combos
+pub trait StreamSink01CompatExt: Stream01 + Sink01 {
```

Comment:
I've used as a reference impl for this https://github.com/google/tarpc/blob/master/bincode-transport/src/compat.rs